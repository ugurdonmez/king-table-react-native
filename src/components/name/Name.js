import React, { Component } from 'react'
import { View, Text, StyleSheet } from 'react-native'
import Icon from '../icon/Icon'

class Name extends Component {
    render() {
        return (

            <View style={styles.container}>
                <Text style={styles.nameText}>ugur</Text>

                <View style={styles.iconContainer}>
                    <Icon style={styles.icon} name="TriangleEmpty" height="15" width="15"/>
                    <Icon style={styles.icon} name="TriangleEmpty" height="15" width="15"/>
                </View>

                <View style={styles.iconContainer}>
                    <Icon style={styles.icon} name="CircleEmpty" height="15" width="15"/>
                    <Icon style={styles.icon} name="CircleEmpty" height="15" width="15"/>
                    <Icon style={styles.icon} name="CircleEmpty" height="15" width="15"/>
                </View>
            </View>
        )
    }
}

export default Name

var styles = StyleSheet.create({
    container: {
        borderRadius: 4,
        borderWidth: 0.5,
        borderColor: '#d6d7da',
        width: '20%',
    },
    iconContainer: {
        flexDirection: 'row',
        justifyContent: 'center',
        borderStyle: 'solid',
    },
    icon: {
        margin: 3,
    },
    nameText: {
        textAlign: 'center',
    }
})