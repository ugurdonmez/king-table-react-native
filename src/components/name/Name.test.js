import React from 'react';
import 'react-native';
import Name from './Name';
import { shallow } from 'enzyme'
import renderer from 'react-test-renderer'
import toJson from 'enzyme-to-json';

describe('<Name /> tests', () => {
    it('renders without crashing', () => {
        shallow(
            <Name 
                name="test"
                punish={2}
                reward={1}
            />)
    })

    it('snapshot test', () => {
        const tree = shallow(
            <Name 
                name="test"
                punish={1}
                reward={1}
            />);

        expect(toJson(tree)).toMatchSnapshot();
    })

    it('shows correct name', () => {
        const wrapper = shallow(
            <Name 
                name="test"
                punish={2}
                reward={1}
            />
        );
    
        wrapper.find('Text').forEach(child => {
            expect(child.children().text()).toBe('test');
        })
    })

    it('should show 2 empty triangle and 3 empty circle', () => {
        const wrapper = shallow(
            <Name 
                name="test"
                punish={0}
                reward={0}
            />
        );

        const components = wrapper.find('Component');

        expect(components.length).toBe(3);

        // second component for reward, two empty triangle
        expect(components.at(1).children().length).toBe(2);

        components.at(1).children().forEach(child => {
            expect(child.prop('name')).toBe('TriangleEmpty')
        })

        // third component for punish, three empty circle 
        expect(components.at(2).children().length).toBe(3);

        components.at(2).children().forEach(child => {
            expect(child.prop('name')).toBe('CircleEmpty')
        })
    })

    it('reward 1, punish 1 ', () => {
        const wrapper = shallow(
            <Name 
                name="test"
                reward={1}
                punish={1}
            />
        );

        const components = wrapper.find('Component');

        expect(components.length).toBe(3);

        // second component for reward
        expect(components.at(1).children().length).toBe(2);

        console.log(components.at(1).children())

        expect(components.at(1).children().at(0).prop('name')).toBe('Triangle')
        expect(components.at(1).children().at(1).prop('name')).toBe('TriangleEmpty')
        
        // third component for punish
        expect(components.at(2).children().length).toBe(3);

        expect(components.at(2).children().at(0).prop('name')).toBe('Circle')
        expect(components.at(2).children().at(1).prop('name')).toBe('CircleEmpty')
        expect(components.at(2).children().at(2).prop('name')).toBe('CircleEmpty')
    })

    it('should show 2 triangle and 3 circle', () => {
        const wrapper = shallow(
            <Name 
                name="test"
                reward={2}
                punish={3}
            />
        );

        const components = wrapper.find('Component');

        expect(components.length).toBe(3);

        // second component for reward, two empty triangle
        expect(components.at(1).children().length).toBe(2);

        components.at(1).children().forEach(child => {
            expect(child.prop('name')).toBe('Triangle')
        })

        // third component for punish, three empty circle 
        expect(components.at(2).children().length).toBe(3);

        components.at(2).children().forEach(child => {
            expect(child.prop('name')).toBe('Circle')
        })
    })
})
