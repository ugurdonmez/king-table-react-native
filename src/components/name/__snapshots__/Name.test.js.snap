// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Name /> tests snapshot test 1`] = `
<Component
  style={
    Object {
      "borderColor": "#d6d7da",
      "borderRadius": 4,
      "borderWidth": 0.5,
      "width": "20%",
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "textAlign": "center",
      }
    }
  >
    test
  </Text>
  <Component
    style={
      Object {
        "borderStyle": "solid",
        "flexDirection": "row",
        "justifyContent": "center",
      }
    }
  >
    <Icon
      height="15"
      key="0"
      name="Triangle"
      style={
        Object {
          "margin": 3,
        }
      }
      width="15"
    />
    <Icon
      height="15"
      key="1"
      name="TriangleEmpty"
      style={
        Object {
          "margin": 3,
        }
      }
      width="15"
    />
  </Component>
  <Component
    style={
      Object {
        "borderStyle": "solid",
        "flexDirection": "row",
        "justifyContent": "center",
      }
    }
  >
    <Icon
      height="15"
      key="0"
      name="Circle"
      style={
        Object {
          "margin": 3,
        }
      }
      width="15"
    />
    <Icon
      height="15"
      key="1"
      name="CircleEmpty"
      style={
        Object {
          "margin": 3,
        }
      }
      width="15"
    />
    <Icon
      height="15"
      key="2"
      name="CircleEmpty"
      style={
        Object {
          "margin": 3,
        }
      }
      width="15"
    />
  </Component>
</Component>
`;
